mindsdb/integrations/handlers/gitlab_handler/gitlab_tables.py

```diff
@@ -0,0 +1,191 @@
+
+        if query.limit:
+            total_results = query.limit.value
+        else:
+            total_results = 20
+
+        issues_kwargs = {}
+        order_by_conditions = {}
+
+        if query.order_by and len(query.order_by) > 0:
+            order_by_conditions["columns"] = []
+            order_by_conditions["ascending"] = []
+
+            for an_order in query.order_by:
+                if an_order.field.parts[0] != "issues":
+                    next
+
+                if an_order.field.parts[1] in ["created", "updated"]:
+                    if issues_kwargs != {}:
+                        raise ValueError(
+                            "Duplicate order conditions found for created/updated"
+                        )
+
+                    issues_kwargs["sort"] = an_order.field.parts[1]
+                    issues_kwargs["direction"] = an_order.direction
+                elif an_order.field.parts[1] in self.get_columns():
+                    order_by_conditions["columns"].append(an_order.field.parts[1])
+
+                    if an_order.direction == "ASC":
+                        order_by_conditions["ascending"].append(True)
+                    else:
+                        order_by_conditions["ascending"].append(False)
+                else:
+                    raise ValueError(
+                        f"Order by unknown column {an_order.field.parts[1]}"
+                    )
+
+        for a_where in conditions:
+            if a_where[1] == "state":
+                if a_where[0] != "=":
+                    raise ValueError("Unsupported where operation for state")
+                if a_where[2] not in ["open", "closed", "all"]:
+                    raise ValueError(
+                        f"Unsupported where argument for state {a_where[2]}"
+                    )
+
+                issues_kwargs["state"] = a_where[2]
+
+                continue
+            if a_where[1] == "labels":
+                if a_where[0] != "=":
+                    raise ValueError("Unsupported where operation for state")
+
+                issues_kwargs["labels"] = a_where[2].split(",")
+
+                continue
+            if a_where[1] in ["assignee", "creator"]:
+                if a_where[0] != "=":
+                    raise ValueError(f"Unsupported where operation for {a_where[1]}")
+
+                issues_kwargs[a_where[1]] = a_where[2]
+            else:
+                raise ValueError(f"Unsupported where argument {a_where[1]}")
+
+        self.handler.connect()
+
+        gitlab_issues_df = pd.DataFrame(columns=self.get_columns())
+
+        start = True
+        while start:
+            try:
+                issues = self.handler.connection.projects.get(
+                    self.handler.repository
+                ).issues.list()
+                for issue in issues:
+
+                    logger.debug(f"Processing issue {issue.iid}")
+
+                    gitlab_issues_df = pd.concat(
```

Comment:
I think is better to have list of dicts (list of issues) and do pd.concat once after this circle. It should work faster.

Reply:
Issue list came with nested format. So keep as it is.